scripts/travis-test.sh

```diff
@@ -8,20 +8,25 @@ set -eu
 export GOGC=400
 export DENDRITE_LINT_DISABLE_GC=1
 
-# Check that the servers build (this is done explicitly because `gb build` can silently fail (exit 0) and then we'd test a stale binary)
-gb build github.com/matrix-org/dendrite/cmd/dendrite-room-server
-gb build github.com/matrix-org/dendrite/cmd/roomserver-integration-tests
-gb build github.com/matrix-org/dendrite/cmd/dendrite-sync-api-server
-gb build github.com/matrix-org/dendrite/cmd/syncserver-integration-tests
-gb build github.com/matrix-org/dendrite/cmd/create-account
-gb build github.com/matrix-org/dendrite/cmd/dendrite-media-api-server
-gb build github.com/matrix-org/dendrite/cmd/mediaapi-integration-tests
-gb build github.com/matrix-org/dendrite/cmd/client-api-proxy
+# We don't bother to build if we're only checking the linting
+if [ "${TEST_SUITE-}" != "lint" ]; then
+    # Check that the servers build (this is done explicitly because `gb build` can silently fail (exit 0) and then we'd test a stale binary)
+    gb build github.com/matrix-org/dendrite/cmd/dendrite-room-server
+    gb build github.com/matrix-org/dendrite/cmd/roomserver-integration-tests
+    gb build github.com/matrix-org/dendrite/cmd/dendrite-sync-api-server
+    gb build github.com/matrix-org/dendrite/cmd/syncserver-integration-tests
+    gb build github.com/matrix-org/dendrite/cmd/create-account
+    gb build github.com/matrix-org/dendrite/cmd/dendrite-media-api-server
+    gb build github.com/matrix-org/dendrite/cmd/mediaapi-integration-tests
+    gb build github.com/matrix-org/dendrite/cmd/client-api-proxy
+fi
 
 # Run unit tests and linters
 ./scripts/build-test-lint.sh
 
-# Run the integration tests
-bin/roomserver-integration-tests
-bin/syncserver-integration-tests
-bin/mediaapi-integration-tests
+if [ "${TEST_SUITE-unit-test}" == "integ-test" ]; then
```

Comment:
why `"${TEST_SUITE-unit-test}"` rather than `"${TEST_SUITE}"` ?

Reply:
Its to correctly handle unset variables without bash being sad (since we have set -eu)